pkg/install/images/images.go

```diff
@@ -238,23 +226,129 @@ func extractAddonsFromArchive(dir string, reader *tar.Reader) error {
 	return nil
 }
 
-func ProcessImages(ctx context.Context, log logrus.FieldLogger, dryRun bool, images []string, registry string, userAgent string) (int, int, error) {
+func ArchiveImages(ctx context.Context, log logrus.FieldLogger, archivePath string, dryRun bool, images []string) (int, int, error) {
+	srcToImage := make(map[string]v1.Image)
+	for _, src := range images {
+		log = log.WithFields(logrus.Fields{
+			"image": src,
+		})
+		log.Info("Fetching image…")
+		img, err := crane.Pull(src, crane.WithAuthFromKeychain(authn.DefaultKeychain), crane.WithContext(ctx))
+		if err != nil {
+			log.WithError(err).Error("Failed to fetch remote image. Skipping...")
```

Comment:
Yeah, I was not sure if we should break because of a few images. This exporter command takes a while and perhaps one can resolve the failing images retroactively.

There could be a reason, besides images being unavailable, why a pull is failing: if the image is of an unsupported type. 
This is actually happening with the image `quay.io/coreos/flannel:v0.11.0` which is of MediaType `application/vnd.docker.distribution.manifest.v1+prettyjws`.

I actually looked into how we can support those. It needs bumping go-containerregistry to 0.16.1 and then some more logic on our side, but should be doable. 

All in all, there could be reasons why it's always failing. Perhaps we can improve the output to clearly mention those images which are not included in the archive and the reason for it. Wdyt?

Reply:
Okay, that's some pretty good points. The count on the last log line should be different, so you would see that some images were not exported successfully, right? I think that would be enough for now.